operators/multiclusterobservability/controllers/multiclusterobservability/grafana.go

```diff
@@ -134,10 +135,8 @@ func GenerateGrafanaDataSource(
 	} else if err != nil {
 		return &ctrl.Result{}, err
 	}
-
-	if !reflect.DeepEqual(grafanaDSFound.Data, dsSecret.Data) {
+	if (grafanaDSFound.Data[datasourceKey] != nil && !reflect.DeepEqual(string(grafanaDSFound.Data[datasourceKey]), string(dsSecret.Data[datasourceKey]))) || grafanaDSFound.Data[datasourceKey] == nil {
```

Comment:
since we're comparing strings, can we just use "==" instead of `reflect` package for performance consideration.

Reply:
better to use `bytes.Equal(s1, s2)` from https://golang.org/pkg/bytes/